lib/vector-core/src/config/mod.rs

```diff
@@ -31,12 +71,16 @@ impl Output {
     /// A default output is one without a port identifier (i.e. not a named output) and the default
     /// output consumers will receive if they declare the component itself as an input.
     pub fn default(ty: DataType) -> Self {
-        Self { port: None, ty }
+        Self {
+            port: None,
+            ty,
+            schema_definition: schema::Definition,
+        }
     }
 
     /// Check if the `Output` is a default output
     pub fn is_default(&self) -> bool {
-        self.port.is_none()
+        self.port.is_none() && self.schema_definition.is_empty()
```

Comment:
Currently this is about whether an output is the "default" in a routing sense (i.e. it's unnamed). Do you see schema definitions coming into play around how events are routed? if not, I'd leave this change out. We could also consider a rename to make its purpose clearer.

Reply:
Good catch, yeah that purpose slipped by me, but looking at it again, it makes sense. I'll see if I can come up with a more descriptive name, or at least expand on the doc comment.

The added check should definitely be removed, given your explanation.

Reply:
@lukesteensen it turns out this function isn't used anywhere currently. I opted to remove it for now, but if you feel differently, let me know, and I'll put it back in a follow-up PR, with a more accurate description/name.